### PR TITLE
Refactor scripts used to set up infrap4d environment

### PR DESCRIPTION
**Pull request by nupuruttarwar**
This is a breaking change.

- Use P4CP_INSTALL for setup_env.sh and copy_config_file.sh. This makes scripts suitable for both development and runtime environment
- Rework the scripts to move common scripts for DPDK and ES2K to scripts/common
- Make all the scripts and config files required for setting up runtime environment part of install tree
- Remove execution permissions for setup_env.sh and os_ver_details.sh as these files are only sourced and not executed
- Correct references of these scripts in documentation

### CODE REVIEW

**Comment by ffoulkes** [comment]
The commit comment is incorrect in saying that IPDK_RECIPE is being replaced with P4CP_INSTALL.

IPDK_RECIPE refers to the local directory to which the `networking-recipe` repository was cloned. P4CP_INSTALL replaces the various references to `./install`, which equivalent to `$IPDK_RECIPE/install`.

File: docs/guides/dpdk-guide.md [hunk 1]
```diff
@@ -69,16 +71,16 @@ appropriate path instead of `./install`.
...
+sudo bash $P4CP_INSTALL/sbin/copy_config_files.sh $P4CP_INSTALL $SDE_INSTALL
 ```
 
 #### Set hugepages required for DPDK
 
 Run the hugepages script.
 
 ```bash
-sudo ./scripts/dpdk/set_hugepages.sh
+sudo bash $P4CP_INSTALL/sbin/set_hugepages.sh
```
Comment (ffoulkes) [inline docs/guides/dpdk-guide.md:83]: Why are you injecting a `bash` command? If set_hugepages.sh has execute permission, you should be able to execute it directly.

This question applies throughout this document.

**Comment by nupuruttarwar** [comment]
> The commit comment is incorrect in saying that IPDK_RECIPE is being replaced with P4CP_INSTALL.
> 
> IPDK_RECIPE refers to the local directory to which the `networking-recipe` repository was cloned. P4CP_INSTALL replaces the various references to `./install`, which equivalent to `$IPDK_RECIPE/install`.
> 
> In fact, the document still has references to IPDK_RECIPE in it.

You are right. I meant to say that about the scripts. Let me change the commit message.

File: docs/guides/dpdk-guide.md (hunk 1 above)
Comment (nupuruttarwar) [reply docs/guides/dpdk-guide.md:83]: I was seeing an error when I didn't have bash. But now when I try this without bash, I don't see the error. Let me update to remove bash

Comment (nupuruttarwar) [reply docs/guides/dpdk-guide.md]: Removed

Comment (ffoulkes) [reply docs/guides/dpdk-guide.md]: > I was seeing an error when I didn't have bash. But now when I try this without bash, 

You might have had a script that didn't have execute permission.

**Review by ffoulkes — APPROVED** [review]
LGTM

File: docs/guides/dpdk-guide.md [hunk 2]
```diff
@@ -95,7 +97,7 @@ certificates, see the [security_guide](https://github.com/ipdk-io/networking-rec
 
 ```bash
 cd $IPDK_RECIPE
```
Comment (jayapalrb) [inline docs/guides/dpdk-guide.md:99]: "cd $IPDK_RECIPE" this step may not be required now.

File: docs/guides/es2k-guide.md [hunk 3]
```diff
@@ -66,15 +68,16 @@ appropriate path instead of ./install.
 we are creating directories and copying config file at system level.
 
 ```bash
-sudo ./scripts/es2k/copy_config_files.sh $IPDK_RECIPE $SDE_INSTALL
+source $P4CP_INSTALL/sbin/setup_env.sh $P4CP_INSTALL $SDE_INSTALL $DEPEND_INSTALL
+sudo bash $P4CP_INSTALL/sbin/copy_config_files.sh $P4CP_INSTALL $SDE_INSTALL
```
Comment (5abeel) [inline docs/guides/es2k-guide.md:72]: Should remove the `bash` command here, similar to the dpdk comment that Derek made yesterday

File: docs/guides/es2k-guide.md [hunk 4]
```diff
@@ -66,15 +68,16 @@ appropriate path instead of ./install.
...
 ```
 
 #### Set hugepages required for ES2K
 
 Run the hugepages script.
 
 ```bash
-sudo ./scripts/es2k/set_hugepages.sh
+sudo bash $P4CP_INSTALL/sbin/set_hugepages.sh
```
Comment (5abeel) [inline docs/guides/es2k-guide.md:80]: same comment. remove `bash`, which starts a new shell process

File: docs/guides/es2k-guide.md [hunk 5]
```diff
@@ -102,7 +105,7 @@ the range with cfgqs-idx parameter. Total number of queues split between process
 
 ```bash
 cd $IPDK_RECIPE
-sudo ./install/sbin/infrap4d
+sudo bash $P4CP_INSTALL/sbin/infrap4d
```
Comment (5abeel) [inline docs/guides/es2k-guide.md:108]: same comment. Remove `bash`

File: scripts/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+# Copyright 2023 Intel Corporation
+# SPDX-License-Identifier: Apache 2.0
+#
+
+add_subdirectory(common)
+
+if(DPDK_TARGET)
+    add_subdirectory(dpdk)
+elseif(ES2K_TARGET)
+    add_subdirectory(es2k)
```
Comment (5abeel) [inline scripts/CMakeLists.txt:12]: Nothing for Tofino target? Might want to add a `TODO` comment if its to be done later

**Review by 5abeel — CHANGES_REQUESTED** [review]

File: scripts/CMakeLists.txt (hunk 6 above)
Comment (ffoulkes) [reply scripts/CMakeLists.txt:12]: The Tofino IPU does not have any scripts. There are no plans to add any.

File: docs/guides/dpdk-guide.md (hunk 2 above)
Comment (nupuruttarwar) [reply docs/guides/dpdk-guide.md:99]: Good catch. thanks

**Review by 5abeel — APPROVED** [review]
LGTM

**Pull request merged**